src/modules/accounting/accounting.py

```diff
@@ -0,0 +1,210 @@
+
+    def _submit_extra_data(self, blockstamp: BlockStamp) -> None:
+        extra_data = self.lido_validator_state_service.get_extra_data(blockstamp, self._get_chain_config(blockstamp))
+
+        tx = self.report_contract.functions.submitReportExtraDataList(extra_data.extra_data)
+
+        if not variables.ACCOUNT:
+            logger.info({'msg': 'No account provided to submit extra data. Dry mode'})
+            return
+        if self.w3.transaction.check_transaction(tx, variables.ACCOUNT.address):
+            self.w3.transaction.sign_and_send_transaction(tx, variables.GAS_LIMIT, variables.ACCOUNT)
+
+    # Consensus module: main build report method
+    @lru_cache(maxsize=1)
+    def build_report(self, blockstamp: BlockStamp) -> tuple:
+        logger.info({'msg': 'Calculate report for accounting module.'})
+        report_data = self._calculate_report(blockstamp)
+        return report_data.as_tuple()
+
+    # # Consensus module: if contract got report data
+    def is_main_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        logger.info({'msg': 'Check if main data was submitted.', 'value': processing_state.main_data_submitted})
+        return processing_state.main_data_submitted
+
+    # Consensus module: if contract could accept any sort of report
+    def is_contract_reportable(self, blockstamp: BlockStamp) -> bool:
+        is_reportable = not self.is_main_data_submitted(blockstamp) or not self.is_extra_data_submitted(blockstamp)
+        logger.info({'msg': 'Check if contract could accept report.', 'value': is_reportable})
+        return is_reportable
+
+    @lru_cache(maxsize=1)
+    def _get_processing_state(self, blockstamp: BlockStamp) -> ProcessingState:
+        ps = named_tuple_to_dataclass(
+            self.report_contract.functions.getProcessingState().call(block_identifier=blockstamp.block_hash),
+            ProcessingState,
+        )
+        logger.info({'msg': 'Fetch processing state.', 'value': ps})
+        return ps
+
+    def is_extra_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        return processing_state.extra_data_items_count == processing_state.extra_data_items_submitted
+
+    def _calculate_report(self, blockstamp: BlockStamp) -> ReportData:
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+
+        finalization_share_rate = self._get_finalization_shares_rate(blockstamp)
+        last_withdrawal_id_to_finalize = self._get_last_withdrawal_request_to_finalize(blockstamp)
+
+        exited_validators = self.lido_validator_state_service.get_lido_new_exited_validators(blockstamp)
+
+        # Here report all exited validators even they were reported before.
+        if exited_validators:
+            staking_module_id_list, exit_validators_count_list = zip(*exited_validators.items())
```

Comment:
`staking_module_id_list` is a `NodeOperatorIndex`, and `exit_validators_count_list` is a count by an operator

but the contract expects:
- stacking_module_id_with_exited_validators - array of module ids
- count_exited_validators_by_stacking_module - array of total exited validators in module